p24/src/refs.rs

```diff
@@ -0,0 +1,39 @@
+pub fn f1(s: &mut (u32, u32), flag: bool) -> &mut u32 {
+    match flag {
+        false => return &mut s.0,
+        true => return &mut s.1,
+    }
+}
+
+// pub fn f2(s: &mut [u32], length: usize) -> &u32 {
+//     let x: &mut [u32];
+//     x = &mut s[length - 1..3];
+//     return &x[1];
+// }
+
+// pub fn f2(s: &mut [u32], length: usize) -> &u32 {
+//     return &mut s[length - 1];
+// }
+
+// pub fn f3(s: &mut [u32], length: usize) -> &u32 {
+//     return &mut s[s.len() - length];
+// }
+
+pub fn f2(s: &mut [u32], length: usize) -> &mut u32 {
+    return &mut s[length - 1];
+}
+
+pub fn f3(s: &mut [u32], length: usize) -> &mut u32 {
+    return &mut s[s.len() - length];
```

Comment:
When you pass `length` equal to zero to this function it should return reference to the last element. This function will panic instead.